src/EventStore.Core/Index/TableIndex.cs

```diff
@@ -222,25 +229,77 @@ public void AddEntries(long commitPos, IList<IndexKey> entries)
+            }
+        }
+     
+        public Task MergeIndexes()
+        {
+            TryManualMerge();
+            return Task.CompletedTask;
+
+            //return Task.Run(() =>
+            //{
+            //    try
+            //    {
+            //        if (_indexMap.CommitCheckpoint == -1 && _indexMap.PrepareCheckpoint == -1) return;
+            //        TryProcessAwaitingTables(_indexMap.CommitCheckpoint, _indexMap.PrepareCheckpoint);
+            //    }
+            //    catch (Exception exc)
+            //    {
+            //        Log.ErrorException(exc, "MERGING INDEXES: error while merging indexes.");
+            //    }
+            //}, CancellationToken.None); 
+        }
+
+        public bool IsBackgroundTaskRunning
+        {
+            get { return _backgroundRunning; }
+        }
 
-                lock (_awaitingTablesLock)
-                {
-                    var newTables = new List<TableItem> { new TableItem(_memTableFactory(), -1, -1) };
-                    newTables.AddRange(_awaitingMemTables.Select(
-                        (x, i) => i == 0 ? new TableItem(x.Table, prepareCheckpoint, commitPos) : x));
+		//Automerge only
+        private void TryProcessAwaitingTables(long commitPos, long prepareCheckpoint)
+        {
+            lock (_awaitingTablesLock)
+            {
+                var newTables = new List<TableItem> {new TableItem(_memTableFactory(), -1, -1, 0)};
+                newTables.AddRange(_awaitingMemTables.Select(
+                    (x, i) => i == 0 ? new TableItem(x.Table, prepareCheckpoint, commitPos, x.Level) : x));
 
                     Log.Trace("Switching MemTable, currently: {awaitingMemTables} awaiting tables.", newTables.Count);
 
-                    _awaitingMemTables = newTables;
-                    if (_inMem) return;
-                    TryProcessAwaitingTables();
+                _awaitingMemTables = newTables;
+                if (_inMem) return;
+                TryProcessAwaitingTables();
 
-                    if (_additionalReclaim)
-                        ThreadPool.QueueUserWorkItem(x => ReclaimMemoryIfNeeded(_awaitingMemTables));
-                }
+                if (_additionalReclaim)
+                    ThreadPool.QueueUserWorkItem(x => ReclaimMemoryIfNeeded(_awaitingMemTables));
             }
         }
 
+        public void TryManualMerge()
+        {
+	        lock (_awaitingTablesLock)
+	        {
+		        
+		        var (maxLevel, highest) = _indexMap.GetTableForManualMerge();
+		        if (highest == null) return; //no work to do
+
+				//These values are actually ignored later as manual merge will never change the checkpoint as no
+				//new entries are added, but they can be helpful to see when the manual merge was called
+				//because of the way the "queue" currently works (LIFO) it should always be the same
+			    var prepare = _indexMap.PrepareCheckpoint;
```

Comment:
Indentation here looks weird given the return above?

Reply:
probably a lack of .editorconfig and my editor liking tabs if there isn't one. will fix